tvm/src/codegen/hlsc/codegen_vhls.h

```diff
@@ -15,7 +15,7 @@
 namespace TVM {
 namespace codegen {
 
-class CodeGenVivadoHLS final : public CodeGenHLSC {
```

Comment:
Supposedly you'd better not inherit from VHLS class. Is SHLS a subset of VHLS? i think no, right? you should probably inherit from CodeGenC class.

Reply:
I agree it's kind of confusing. SHLS is not conceptually a subset of VHLS, but they do share the same arithmetic codegen functions. The SHLS class inherits from VHLS to reuse those functions, for example, for loops codegen. Inheriting from CodeGenC class would result in code duplication. For that purpose I think it makes sense to inherit from VHLS